src/Microsoft.DotNet.CodeFormatting/Rules/AttributeNoParenthesesRule.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.DotNet.CodeFormatting.Rules
+{
+    [SyntaxRuleOrder(SyntaxRuleOrder.AttributeNoParenthesesRule)]
+    internal sealed class AttributeNoParenthesesRule : ISyntaxFormattingRule
+    {
+        public SyntaxNode Process(SyntaxNode syntaxRoot)
+        {
+            var attributes = syntaxRoot.DescendantNodes()
+                                       .OfType<AttributeSyntax>()
+                                       .Where(a => a.ArgumentList != null &&
+                                                   a.ArgumentList.Arguments.Count == 0 &&
+                                                   (!a.ArgumentList.OpenParenToken.IsMissing || !a.ArgumentList.CloseParenToken.IsMissing));
+
+            return syntaxRoot.ReplaceNodes(attributes, (a, n) => a.WithArgumentList(null));
```

Comment:
Also I would use 

```
a.WithArgumentList(SyntaxFactory.ArgumentList())
```

I couldn't find any reference as to whether or not that property guaranteed non-nullness but it seems to for well formed code.